### DISCUSSION TITLE
PARC to  CHM

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#1207] [General]
Please share your ideas and discoveries 
@Interlisp

**Comment by larsbrinkhoff** [comment]
What I have *not* discovered yet, is Interlisp compiler support for MAXC ByteLisp instructions.  But I haven't finished downloading all Lisp directories yet.

**Comment by nbriggs** [comment]
I've loaded Trillium from sources, but some updates are needed to actually get it running.  Koto to Medley differences, I think.

**Comment by masinter** [comment]
@larsbrinkhoff What I remember (with some uncertainty of course) is that the Maxc bytelisp implementation was just me and Ed Fiala: he did the microcode and I made the compiler changes. The bytecompiler produced "codes" for an abstract stack machine (see https://dl.acm.org/doi/pdf/10.1145/800087.802810) for a description. The distinction between altolisp/D-machine bytecodes and Maxc 9-bit byte codes was all handled in the last phase of the compiler. 

But because this was just a two-person project without any code-sharing, we might not have put the code up on any shared account. And I don't think they included any personal files in the backups made available, so you might not find the code.

The experiment wasn't successful because loading microcode was a slow process, so you had to fit in both the lisp and the PDP-10 implementation into the (limited) microcode, and things like function and subr calls, process switches, which would switch between lisp bytecode to pdp10 mode which read or write all the registers to memory... was too slow. I think (my memory registers not saved so well).

**Reply by larsbrinkhoff** [reply]
Thank you!  FYI, the Maxc microcode is now available here: https://xeroxparcarchive.computerhistory.org/indigo/maxc/maxc12microcode.dm!3_/.index.html
And it includes the bytelisp implementation.

**Comment by lassik** [comment]
Not sure whether this thread is a good place to ask, but I'd be interested in a mirror of **ftp://ftp.parc.xerox.com/**.

The original FTP site has vanished, and Archive.org has saved [less than 50 files](https://web.archive.org/web/*/ftp://ftp.parc.xerox.com/*).

**Comment by masinter** [comment]
@lassik https://github.com/masinter/parcftp-cl has > 1200 files just from the 'cl' subdirectory from (I think) the archive.org 2014 scan of ftp.parc.xerox.com.... were you looking for those files or something else?

**Comment by lassik** [comment]
Great! Thank you for archiving them.

I'm mostly looking for Lisp and Smalltalk stuff. There was also a `/pub/mops` directory (meta-object protocols).

Is the full 2014 scan available somewhere?

**Comment by masinter** [comment]
I got it from archive.org but now I don't see how to access more
https://drive.google.com/drive/folders/1hf5p6Ss2gjuv0LJtbmaCRA64Tpwj8_BA?usp=sharing

**Comment by lassik** [comment]
It's probably this one: https://archive.org/details/2014.01.ftp.parc.xerox.com

Here is a less complete one from 2015: https://archive.org/details/2015-02-10.ftp.parc.xerox.com
